### PR TITLE
docs: add pull request template & CONTRIBUTING.md

### LINKED ISSUE #1: Add PR Template
A request has been made to create a template to guide developers in the creation of Pull Requests. This template should include a PR naming guide, what requirements or issues its closing, how its been tested, etc.

### LINKED ISSUE #3: Add Contributing.md guidelines
Add a CONTRIBUTING.md file which defines the standards that we're following on Git. Should include the following:
- [ ] Issue Creation 
- [ ] Branch naming
- [ ] Commit message standards
- [ ] Pull request process (opening a PR, filling out the template, reviewing, merging)
- [ ] Release

### PR DESCRIPTION
**Pull request by LoganParker**
## Summary

Closes #1 and #3 by adding the PR template and CONTRIBUTING.md. This is an initial draft of both documents that  contain some simple guidance and comments, as well as document our Github processes. This template was used to create this PR

## Test plan

No relevant tests for this PR. Review & make suggestions to wording, sections, etc. 

## Reviewer notes

Some of these sections may not be necessary / relevant. The comments within the PR template make it somewhat challenging to tell exactly what is happening where, but they do provide relevant info especially for a first time user of the template. I'm working on creating a PR Review Checklist as well as a Contributing.md document that would help to add some additional documentation, at which point the comments may be able to be reduced/removed.

### CODE REVIEW

File: pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+<!--
+What changed and why. Lead with closing keywords, a one-sentence overview, then add detail.
+- Bug fix: describe the symptom, the root cause, and the fix.
+- Feature: describe the new behavior and why it was added.
+- Multi-issue PR: use a sub-bullet per issue, e.g.:
+  - ** Closes #202** — Fixed date format on invoices
+  - ** Closes #203** — Removed trailing comma from totals row
+
+For Firmware changes or changes involving some form of hardware, include the following details:
```
Comment (AdamShier1) [inline pull_request_template.md]: I dont think this specific stuff is necessary, I would just include what you have above.

File: pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+review. Do not request a review on unverified AI-generated code. 
+-->
+
+- [ ] **None**: All code written by the author without AI assistance
+- [ ] **Partial**: Some code or logic was AI-generated. 
+- [ ] **Majority**: Most of the implementation was AI-generated.
+
+## Test plan
+
+<!--
```
Comment (AdamShier1) [inline pull_request_template.md:48]: I think this section should also list what testing has already been done by the PR creator. Repeating the exact same tests the reviewer has done isn't as useful as seeing what theyve done and thinking about different edge cases or things they may have missed.

File: pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
+## Summary
+
+<!--
+What changed and why. Lead with closing keywords, a one-sentence overview, then add detail.
+- Bug fix: describe the symptom, the root cause, and the fix.
+- Feature: describe the new behavior and why it was added.
+- Multi-issue PR: use a sub-bullet per issue, e.g.:
+  - ** Closes #202** — Fixed date format on invoices
+  - ** Closes #203** — Removed trailing comma from totals row
```
Comment (AdamShier1) [inline pull_request_template.md]: I think there should be a top level thing that lists which requirement or specification this PR is related to. Ideally the PR is closing off a requirement or specification.

File: pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+- [ ] **None**: All code written by the author without AI assistance
+- [ ] **Partial**: Some code or logic was AI-generated. 
+- [ ] **Majority**: Most of the implementation was AI-generated.
+
+## Test plan
+
+<!--
+List the steps a reviewer should take to verify this works.
+Be specific: include role (admin vs. non-admin), device, OS, or firmware target where relevant.
```
Comment (AdamShier1) [inline pull_request_template.md]: This section could also reference which unit tests were created to cover this PR. I think ideally the person would add unit tests at the same time as the PR and they would be passing before merging.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,345 @@
...
+- One logical change per commit. Ideally squash WIP / cleanup commits before requesting review
+
+---
+
+## Opening a pull request
+
+### Before you open
+
+- [ ] Work has been committed and pushed regularly throughout development
+- [ ] Branch is up to date with `main` (`git fetch && git rebase origin/main`)
```
Comment (AdamShier1) [inline CONTRIBUTING.md]: The up to date with main is interesting, not sure thats always necessary.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,345 @@
...
+---
+
+### Branch structure
+
+Every repository requires two long-lived branches:
+
+| Branch    | Purpose                                                   |
+|-----------|-----------------------------------------------------------|
+| `release` | Always points to the most recent formal release tag.      |
+| `main`    | Active development branch. All feature work targets here. |
```
Comment (AdamShier1) [inline CONTRIBUTING.md:68]: This is just a preference of mine but i actually prefer release and develop. We can have an internal vote on that though

File: CONTRIBUTING.md (hunk 5 above)
Comment (LoganParker) [reply CONTRIBUTING.md]: Yeah it's mostly just to make the review process go smoother, if you have out of date code it's harder to review the specific changes made in the PR because changes you'd expect to exist based on whats in main might be missing or something. But not necessarily always necessary either. I think this is also covered by the "No unresolved merge conflicts" so i'll just remove this requirement

File: CONTRIBUTING.md (hunk 6 above)
Comment (LoganParker) [reply CONTRIBUTING.md:68]: Haha oh man. I definitely prefer main just from a simplicity standpoint but yes an internal vote sounds like a good plan

File: pull_request_template.md (hunk 2 above)
Comment (LoganParker) [reply pull_request_template.md:48]: Yeah good call. I kinda tried to describe that, based on "check off each step as you confirm it locally before requesting review. Reviewers should then be able to follow this plan to replicate" 

but it's a little vague, so i'll modify the wording there

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,344 @@
...
+
+| Component | When to increment                                                                                  |
+|-----------|----------------------------------------------------------------------------------------------------|
+| **Major** | Breaking change - the software is no longer compatible with connected devices or dependent systems |
+| **Minor** | New features added in a backwards-compatible way                                                   |
+| **Patch** | Backwards-compatible bug fixes only                                                                |
+
+Internal working copies append a **Build** number: `Major.Minor.Patch.Build` (e.g. `1.2.1.16`). The build number is 
+removed on formal release - `1.2.1.16` becomes `1.2.2` (or whichever increment is appropriate for the changes).
+
```
Comment (AdamShier1) [inline CONTRIBUTING.md:303]: Can you add 1 sentence saying it can be like v1.2.3-rc1 for release candidates

**Review by AdamShier1 — APPROVED** [review]

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,344 @@
...
+All releases use semantic versioning: `Major.Minor.Patch`
+
+| Component | When to increment                                                                                  |
+|-----------|----------------------------------------------------------------------------------------------------|
+| **Major** | Breaking change - the software is no longer compatible with connected devices or dependent systems |
+| **Minor** | New features added in a backwards-compatible way                                                   |
+| **Patch** | Backwards-compatible bug fixes only                                                                |
+
+Internal working copies append a **Build** number: `Major.Minor.Patch.Build` (e.g. `1.2.1.16`). The build number is
```
Comment (LoganParker) [inline CONTRIBUTING.md]:
```suggestion
Internal working copies append a **Build** number: `Major.Minor.Patch.Build` (e.g. `1.2.1.16` or 1.2.1-rc16). The build number is 
```
